test/allow.js

```diff
@@ -0,0 +1,131 @@
+const assert = require('chai').assert
+const { getUserId, verifyDelegator } = require('../lib/handlers/allow')
+const rdf = require('rdflib')
+const ns = require('solid-namespace')(rdf)
+
+const alice = 'https://alice.com'  // principal
+const agentWebId = 'https://agent.example.com'  // secretary
+
+const verifiedSecretaryGraph = rdf.graph()
+verifiedSecretaryGraph.add(
+    rdf.namedNode(alice),
+    ns.acl('delegates'),
+    rdf.namedNode(agentWebId)
+  )
+const notVerifiedSecretaryGraph = rdf.graph()
+
+describe('allow() handler', () => {
+  describe('getUserId()', () => {
+    const emptyFetchDocument = () => {}
+    it('no webId present', done => {
```

Comment:
I have a strong preference for test cases to be named after the behavior that they're looking for rather than the data they accept. For example, rather than "no webId present" something like "should <do expected behavior> when no webId is present". This clarifies the intended logic for contributors, future developers, and us months down the line when we forget what the intended behavior was.